server/controllers/hooks.controller.js

```diff
@@ -5,20 +5,57 @@ class HooksController {
     this.shopifyClientFactory = shopifyClientFactory;
   }
 
+  async safeGetShopAsync(redis, shopDomain) {
+    try {
+      return await redis.getShopAsync(shopDomain);
+    } catch (error) {
+      console.error(`Error loading shop ${shopDomain} from Redis DB`, error);
+      return null;
+    }
+  }
+
+  async safeRemoveShopAsync(redis, shopDomain) {
+    try {
+      await redis.removeShopAsync(shopDomain);
+    } catch (error) {
+      console.error(`Error removing shop ${shopDomain} from Redis DB`, error);
+    }
+  }
+
+  async safeRemoveDopplerIntegrationAsync(shop) {
+    if (!shop || !shop.dopplerAccountName || !shop.dopplerApiKey) {
+      console.error('No enough data to remove Doppler integration.', { shop });
+      return;
+    }
+
+    try {
+      const doppler = this.dopplerClientFactory.createClient(shop.dopplerAccountName, shop.dopplerApiKey);
+      await doppler.deleteShopifyIntegrationAsync();
+    } catch (error) {
+      console.error(`Error removing integration from Doppler`, error);
+    }
+  }
+
   async appUninstalled(error, request) {
     if (error) {
-      console.error(error);
+      console.error('appUninstalled hook', error);
       return;
     }
 
-    try
-    {
-      const shopDomain = request.webhook.shopDomain;
+    const shopDomain = request.webhook && request.webhook.shopDomain;
 
-      const redis = this.redisClientFactory.createClient();
-      await redis.removeShopAsync(shopDomain, true);
-    } catch (err) {
-      console.error(err);
+    if (!shopDomain) {
+      console.error('Shop domain not found in the request, cannot cleanup our data.')
```

Comment:
When shopDomain is not found in the request, the ejecution flow must to continue or it should cut here?

Reply:
If I don't have the `shopDomain` I cannot know what to cleanup, so, I cut the flow here.

Reply:
Oh, You are right, Thanks @leoslopez  and @fcoronelmakingsense for the heads up.